### ISSUE TITLE
Possibility to test several examples?

### ISSUE THREAD

**Issue by xballiet** [einaregilsson/Redirector#237]
Would it be possible to have several examples for a rule and to see directly their conversion?

**Comment by grahamperrin** [comment]
### Reference

From https://github.com/einaregilsson/Redirector/blob/master/help.html

> … To test your regular expressions, you may use any website or service. For example, https://regexr.com …

– any good for you? 

### The extension

Whilst I can see the use case for multiple examples, I wonder whether it might over-complicate the (advanced) GUI.

**Comment by xballiet** [comment]
Thanks for your answer!
It is not only a matter of testing the examples, but it is also to keep a bunch of examples when needed.  ;)
In fact I do not use any regexp for now in Redirector (not that I do not know how to write a regexp, but I do not need any for now here), only wildcards.
As for the GUI, we could use semi-columns, commas or carriage returns / line feeds to separate examples (or anything else that would be easily understandable for us and implementable for you) and you could separate the results with carriage returns / line feeds for readability.

**Comment by einaregilsson** [comment]
I think that would complicate the GUI, which I'm not a fan of. I would consider it if this issue gets a significant number of upvotes, but if it doesn't I don't feel like it's worth complicating the code and GUI for.

**Comment by xballiet** [comment]
OK, I upvote my own ticket, but I have the feeling that I will be the only one...  ;)

**Comment by Gitoffthelawn** [comment]
@xballiet Yours is not the only upvote!  I upvoted as well, as I have wanted this for similar reasons.

I use complex regex's in Redirector, and they match many different URL variants.  As such, I've always wanted to have multiple URL examples for each rule.

What I currently do is a bit ugly: I maintain a separate text file with URLs to test.  Yep, it's ugly and time consuming, but it's error-free (unless I goof while making many repetitive copy and paste operations).

To keep the GUI visually simple, we could make the `Example` field an editable drop-down selector while editing a rule (like this: https://stackoverflow.com/questions/264640/how-can-i-create-an-editable-dropdownlist-in-html#17658653).

The `Examples result` field would simply show the result for the selected example.

Visually, this would not change the main UI at all, and would only minimally change the edit UI.

Alternatively, a simple numeric dropdown selector could be placed next to the `Example` field in the edit UI.  It will select the current example to use.  If we think that adds too much complexity, it could easily be toggled by an advanced setting.

In both techniques, the main Redirector UI is not altered at all.

**Comment by Gitoffthelawn** [comment]
@deadmeu Yes. It just requires someone to volunteer to write the code to implement the feature.
